### ISSUE TITLE
Progress bar rendering on Windows broken

### ISSUE THREAD

**Issue by thers** [thephpleague/climate#9]
I'm using ANSICON (because default terminal don't support ANSI escape codes, in fact ANSICON don't do it well too) on Win8x64, CLImate just renders new line each time it increments, without erasing old one.

So it looks like this:

```
>
=>
==>
===>
====>
=====>
======>
=======>
========>
=========>
==========>
===========>
============>
=============>
==============>
===============>
================>
=================>
==================>
===================>
====================>
=====================>
======================>
=======================>
========================>
=========================>
==========================>
===========================>
============================>
=============================>
==============================>
===============================>
================================>
=================================>
==================================>
===================================>
====================================>
                         n%
```

Well, i know that it's a Windows problem, but look at Composer, it renders progress bar fine even on Windows ;)

**Comment by vkbansal** [comment]
Getting similar warning on windows
```
Warning: str_repeat(): Second argument has to be greater than or equal to 0 in D:\repos\vendor\league\climate\src\TerminalObject\Dynamic\Progress.php on line 158
```

**Comment by sergeizelenyi** [comment]
my output comand mode in cmd:
Status for device CON:
----------------------
    Lines:          3000
    Columns:        238
    Keyboard rate:  31
    Keyboard delay: 1
    Code page:      866
https://github.com/thephpleague/climate/blob/master/src/Util/System/Windows.php#L14
return $this->getDimension('width');

but value width contains in  Columns
if return $this->getDimension('columns'), work its ok

Google translate :(

**Comment by nyamsprod** [comment]
What's up with this bug ? I tried today climate and had the same problem this times with a Linux build from Travis-CI

**Comment by joetannenbaum** [comment]
I'll take a look at this tonight when I get home and get back to you. I have an idea for an alternative method to re-write the line, I'll check it out and see if it works.

**Comment by joetannenbaum** [comment]
Well, after a lot of unsuccessful tinkering tonight, no change in the result unfortunately.

@nyamsprod when it's run outside of Travis it looks ok? I think Travis is having trouble displaying the output because the progress bar is asking the cursor to move up, not just re-writing the same line over and over.

@vkbansal what are you running the command on in Windows? And what version of CLImate are you running?

**Comment by vkbansal** [comment]
@joetannenbaum I'm running commands via ConEmu (x64) and the version I used was `2.6.1`

**Comment by RiderSx** [comment]
I have no opportunity  to test this on Win8, but now I can test it on Win10TP, I'll drop a word about the results later :)

**Comment by sergeizelenyi** [comment]
Warning: str_repeat(): Second argument has to be greater than or equal to 0 in C:\xampp\htdocs\dbcompare\vendor\league\climate\src\TerminalObject\Dynamic\Progress.php on line 160

fix:
my output 'mode' on Win7
Status for device COM1:
-----------------------
Baud:            1200
Parity:          None
Data Bits:       7
Stop Bits:       1
Timeout:         OFF
XON/XOFF:        OFF
CTS handshaking: OFF
DSR handshaking: OFF
DSR sensitivity: OFF
DTR circuit:     ON
RTS circuit:     ON


Status for device CON:
----------------------
Lines:          3000
Columns:        238
Keyboard rate:  31
Keyboard delay: 1
Code page:      866

https://github.com/thephpleague/climate/blob/master/src/Util/System/Windows.php#L58
$matches[1]
array(8) {
[0] =>
string(4) "1200"
[1] =>
string(1) "7"
[2] =>
string(1) "1"
[3] =>
string(4) "3000"
[4] =>
string(3) "238"
[5] =>
string(2) "31"
[6] =>
string(1) "1"
[7] =>
string(3) "866"
} 

https://github.com/thephpleague/climate/blob/master/src/Util/System/Windows.php#L37

$index  = array_search($key, ['height', 'width']);
index 0 and 1
matches 1200 and 7

https://github.com/thephpleague/climate/blob/master/src/TerminalObject/Dynamic/Progress.php#L174
 min($this->util->width() - 10, 100)
 min(7 - 10, 100) = -3

Fix:
1. abs(min($this->util->width() - 10, 100))
2. in current $matches return index 3,4

**Comment by yparisien** [comment]
The solution is to 
change this line in src/Util/System/Windows.php

`$output = $this->exec('mode', true);`

by this one 

`$output = $this->exec('mode CON', true);`

**Comment by lapin-b** [comment]
I've tested with Cmder and nothing is broken. I'm on Windows 8.1 using PHP 5.5.12

**Comment by mjordan** [comment]
Tested in Windows 7 Enterprise SP1 (64 bit) using Git Bash, cmd, and PowerShell, with the same behavior as illustrated by @RiderSx.

**Comment by borgogelli** [comment]
I use ConEmu and I see a similar bug with the padding method that is related to a wrong width size detection of the terminal window. I have solved by the suggestion of  @yparisien.
Regards from Italy

**Comment by duncan3dc** [comment]
We've merged #64 now, as it looks like this issue contains multiple separate problems, please test out `dev-master` and open a new issue for any remaining problems, thanks
